quickwit/quickwit-control-plane/src/control_plane.rs

```diff
@@ -253,8 +253,8 @@ impl Handler<DeleteIndexRequest> for ControlPlane {
     ) -> Result<Self::Reply, ActorExitStatus> {
         let index_uid: IndexUid = request.index_uid.clone().into();
 
-        if let Err(error) = self.metastore.delete_index(request).await {
-            return Ok(Err(ControlPlaneError::from(error)));
+        if let Err(metastore_error) = self.metastore.delete_index(request).await {
+            return convert_metastore_error(metastore_error);
```

Comment:
that's a bugfix I believe. Without this, if we encounterred an error from the metastore, depending on the error, we could observe a divergence between the control plane model  and the metastore.

Reply:
We should add a test on this I think. Can you open an issue and assigned it to me?

Reply:
created #4185